drivers/sensor/adi/adxl345/adxl345.c

```diff
@@ -54,60 +54,239 @@ static int adxl345_reg_access_spi(const struct device *dev, uint8_t cmd, uint8_t
 		.count = 2,
 	};
 
+	int ret = 2;
 	if (cmd == ADXL345_READ_CMD) {
 		tx.count = 1;
-		return spi_transceive_dt(&cfg->bus.spi, &tx, &rx);
+		ret = spi_transceive_dt(&cfg->bus.spi, &tx, &rx);
+		return ret;
 	} else {
-		return spi_write_dt(&cfg->bus.spi, &tx);
+		ret = spi_write_dt(&cfg->bus.spi, &tx);
+		return ret;
 	}
 }
 #endif /* DT_ANY_INST_ON_BUS_STATUS_OKAY(spi) */
 
-static inline int adxl345_reg_access(const struct device *dev, uint8_t cmd, uint8_t addr,
+int adxl345_reg_access(const struct device *dev, uint8_t cmd, uint8_t addr,
 				     uint8_t *data, size_t len)
 {
 	const struct adxl345_dev_config *cfg = dev->config;
 
 	return cfg->reg_access(dev, cmd, addr, data, len);
 }
 
-static inline int adxl345_reg_write(const struct device *dev, uint8_t addr, uint8_t *data,
+int adxl345_reg_write(const struct device *dev, uint8_t addr, uint8_t *data,
 				    uint8_t len)
 {
 
 	return adxl345_reg_access(dev, ADXL345_WRITE_CMD, addr, data, len);
 }
 
-static inline int adxl345_reg_read(const struct device *dev, uint8_t addr, uint8_t *data,
+int adxl345_reg_read(const struct device *dev, uint8_t addr, uint8_t *data,
 				   uint8_t len)
 {
 
 	return adxl345_reg_access(dev, ADXL345_READ_CMD, addr, data, len);
 }
 
-static inline int adxl345_reg_write_byte(const struct device *dev, uint8_t addr, uint8_t val)
+int adxl345_reg_write_byte(const struct device *dev, uint8_t addr, uint8_t val)
 {
 	return adxl345_reg_write(dev, addr, &val, 1);
 }
 
-static inline int adxl345_reg_read_byte(const struct device *dev, uint8_t addr, uint8_t *buf)
+int adxl345_reg_read_byte(const struct device *dev, uint8_t addr, uint8_t *buf)
 
 {
 	return adxl345_reg_read(dev, addr, buf, 1);
 }
 
+int adxl345_reg_write_mask(const struct device *dev,
+			       uint8_t reg_addr,
+			       uint32_t mask,
+			       uint8_t data)
```

Comment:
Why is the mask wider than the data?

Reply:
It's now set to uint8_t mask, It was my oversight on my part.